### PR TITLE
RenderGraph: add package.

### PR DESCRIPTION
**Pull request by DragonJoker**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/r/rendergraph/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            add_rules("mode.debug", "mode.release")
+            add_requires("vulkan-headers")
+            target("RenderGraph")
+                set_kind("shared")
+                add_includedirs("include")
+                add_files("source/RenderGraph/RunnablePasses/*.cpp")
+                add_files("source/RenderGraph/*.cpp")
+                set_languages("c++20")
+                add_defines("RenderGraph_EXPORTS")
+                add_headerfiles("include/RenderGraph/RunnablePasses/*.hpp", {prefixdir="RenderGraph/RunnablePasses"})
```
Comment (waruqi) [inline packages/r/rendergraph/xmake.lua:24]: add spaces. `prefixdir = "`

File: packages/r/rendergraph/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    add_deps("vulkan-headers")
+
+    add_links("RenderGraph")
+
+    on_install("windows|x64", "macosx", "linux", function (package)
+        io.writefile("xmake.lua", [[
+            add_rules("mode.debug", "mode.release")
+            add_requires("vulkan-headers")
+            target("RenderGraph")
+                set_kind("shared")
```
Comment (waruqi) [inline packages/r/rendergraph/xmake.lua:18]: set_kind("$(kind)")

File: packages/r/rendergraph/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                add_includedirs("include")
+                add_files("source/RenderGraph/RunnablePasses/*.cpp")
+                add_files("source/RenderGraph/*.cpp")
+                set_languages("c++20")
+                add_defines("RenderGraph_EXPORTS")
+                add_headerfiles("include/RenderGraph/RunnablePasses/*.hpp", {prefixdir="RenderGraph/RunnablePasses"})
+                add_headerfiles("include/RenderGraph/*.hpp", {prefixdir="RenderGraph"})
+                add_packages("vulkan-headers")
+        ]])
```
Comment (waruqi) [inline packages/r/rendergraph/xmake.lua:27]: add configs to switch static/shared kind. see https://github.com/xmake-io/xmake-repo/blob/f574cfe3aeb21fa157155b907a5a037b17f5ff5e/packages/l/libpng/xmake.lua#L48-L49

File: packages/r/rendergraph/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        io.writefile("xmake.lua", [[
+            add_rules("mode.debug", "mode.release")
+            add_requires("vulkan-headers")
+            target("RenderGraph")
+                set_kind("shared")
+                add_includedirs("include")
+                add_files("source/RenderGraph/RunnablePasses/*.cpp")
+                add_files("source/RenderGraph/*.cpp")
+                set_languages("c++20")
+                add_defines("RenderGraph_EXPORTS")
```
Comment (waruqi) [inline packages/r/rendergraph/xmake.lua:23]:
```lua
if is_kind("shared") then
    add_defines("RenderGraph_EXPORTS")
end
```

File: packages/r/rendergraph/xmake.lua (hunk 3 above)
Comment (DragonJoker) [reply packages/r/rendergraph/xmake.lua:27]: So, in package installation :
```
if package:config("shared") then 
    configs.kind = "shared"
end
```
In the generated xmake.lua :
```
set_kind("$(kind)")
```
Right ?

Comment (waruqi) [reply packages/r/rendergraph/xmake.lua:33]: right

File: packages/r/rendergraph/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                set_languages("c++20")
+                if is_plat("windows") then
+                    if is_kind("shared") then
+                        add_defines("RenderGraph_EXPORTS")
+                    else
+                        add_defines("CRG_BUILD_STATIC")
+                    end
+                end
+                add_headerfiles("include/RenderGraph/RunnablePasses/*.hpp", {prefixdir = "RenderGraph/RunnablePasses"})
+                add_headerfiles("include/RenderGraph/*.hpp", {prefixdir = "RenderGraph"})
```
Comment (waruqi) [inline packages/r/rendergraph/xmake.lua:31]: only need

```lua
add_headerfiles("include/(RenderGraph/**.hpp)")
```

File: packages/r/rendergraph/xmake.lua [hunk 6]
```diff
@@ -0,0 +1,52 @@
+package("rendergraph")
+
+    set_homepage("https://github.com/DragonJoker/RenderGraph/")
+    set_description("Vulkan render graph management library. .")
+
+    set_urls("https://github.com/DragonJoker/RenderGraph.git")
```
Comment (waruqi) [inline packages/r/rendergraph/xmake.lua:6]: Add a tar url if it exists

```lua
add_urls("https://github.com/DragonJoker/RenderGraph/archive/refs/tags/$(version).tar.gz",
                "https://github.com/DragonJoker/RenderGraph.git")
add_versions("v1.0.0", "<tar.gz sha256>")
```

Comment (DragonJoker) [reply packages/r/rendergraph/xmake.lua:6]: Well, the commit is not on the tag

File: packages/r/rendergraph/xmake.lua [hunk 7]
```diff
@@ -0,0 +1,51 @@
+package("rendergraph")
+
+    set_homepage("https://github.com/DragonJoker/RenderGraph/")
+    set_description("Vulkan render graph management library. .")
```
Comment (waruqi) [inline packages/r/rendergraph/xmake.lua:4]: set_license("MIT")

File: packages/r/rendergraph/xmake.lua [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+    on_install("windows|x64", "macosx", "linux", function (package)
+        io.writefile("xmake.lua", [[
+            add_rules("mode.debug", "mode.release")
+            add_requires("vulkan-headers")
+            target("RenderGraph")
+                set_kind("$(kind)")
+                add_includedirs("include")
+                add_files("source/RenderGraph/RunnablePasses/*.cpp")
+                add_files("source/RenderGraph/*.cpp")
```
Comment (waruqi) [inline packages/r/rendergraph/xmake.lua:21]: only need 
```lua
add_files("source/RenderGraph/**.cpp")
```

File: packages/r/rendergraph/xmake.lua (hunk 6 above)
Comment (waruqi) [reply packages/r/rendergraph/xmake.lua:6]: you can a fake version.
```lua
add_urls("https://github.com/DragonJoker/RenderGraph/archive/refs/tags/$(version).tar.gz",
         "https://github.com/DragonJoker/RenderGraph.git")
add_versions("v1.0.0", "<tar.gz sha256>")
add_versions("v1.0.0-fix", "git commit")
```

Comment (DragonJoker) [reply packages/r/rendergraph/xmake.lua]: Given the change from the tag, I've picked it instead of the commit ID.

**Pull request merged**